runtime/vm/ContinuationHelpers.cpp

```diff
@@ -1239,6 +1239,16 @@ takeVirtualThreadListToUnblock(J9VMThread *currentThread)
 				case JAVA_LANG_VIRTUALTHREAD_TIMED_WAIT:
 					/* WAIT/TIMED_WAIT can only be added to blocked list if they have been notified. */
 					Assert_VM_true(J9VMJAVALANGVIRTUALTHREAD_NOTIFIED(currentThread, current->vthread));
+					/* The transition to BLOCKED may have been missed if vthread was in a WAITING state while notified. */
+					if (MM_ObjectAccessBarrierAPI(currentThread).inlineMixedObjectCompareAndSwapU32(
```

Comment:
This function has a default sixth parameter `bool isVolatile = false` which need to be set in this case since state is declared as `private volatile int state;`
see https://github.com/eclipse-openj9/openj9/blob/master/runtime/gc_include/ObjectAccessBarrierAPI.hpp#L726-L741